src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,560 @@
+                        conditionsMet = conditionsMet && parameter.value > condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN:
+                        conditionsMet = conditionsMet && parameter.value < condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_GREATER_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value >= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value <= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value === condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_NOT_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value !== condition.value;
+                        break;
+                }
+                if (!conditionsMet)
+                    return conditionsMet;
+            }
+            return conditionsMet;
+        },
+        hasExitTime: function () {
+            return !!this.exitTime;
+        }
+    });
+
+    var AnimController = function (animEvaluator, states, transitions, parameters, activate) {
+        this.animEvaluator = animEvaluator;
+        this.states = {};
+        var i;
+        for (i = 0; i < states.length; i++) {
+            this.states[states[i].name] = new AnimState(
+                states[i].name,
+                states[i].speed
+            );
+        }
+        this.transitions = transitions.map(function (transition) {
+            return new AnimTransition(
+                this,
+                transition.from,
+                transition.to,
+                transition.time,
+                transition.priority,
+                transition.conditions,
+                transition.exitTime,
+                transition.transitionOffset,
+                transition.interruptionSource
+            );
+        }.bind(this));
+        this.findTransitionsFromStateCache = {};
+        this.findTransitionsBetweenStatesCache = {};
+        this.parameters = parameters;
+        this.initialParameters = Object.assign({}, parameters);
+        this.previousStateName = null;
+        this.activeStateName = ANIM_STATE_START;
+        this.playing = false;
+        this.activate = activate;
+
+        this.currTransitionTime = 1.0;
+        this.totalTransitionTime = 1.0;
+        this.isTransitioning = false;
+        this.transitionInterruptionSource = ANIM_INTERRUPTION_SOURCE_NONE;
+        this.transitionPreviousStates = [];
+
+        this.timeInState = 0;
+        this.timeInStateBefore = 0;
+    };
+
+    Object.assign(AnimController.prototype, {
+        _getState: function (stateName) {
```

Comment:
These should be JavaScript properties

Reply:
This is more of a find than a get so i've updated the naming.